src/routes/safe/components/Transactions/TxsTable/ExpandedTx/TxDescription/CustomDescription.tsx

```diff
@@ -0,0 +1,183 @@
+                  </InlineText>
+                  <Value method={methodName} type={param.type} value={param.value} />
+                </TxDetailsMethodParam>
+              ))}
+            </TxInfo>
+          )}
+        </TxDetailsContent>
+      </Collapse>
+    </>
+  )
+}
+
+const TxData = ({ data }: { data: string }): React.ReactElement => {
+  const classes = useStyles()
+  const [showTxData, setShowTxData] = React.useState(false)
+  const showExpandBtn = data.length > 20
+
+  return (
+    <Paragraph className={classes.txDataParagraph} noMargin size="md">
+      {showExpandBtn ? (
+        <>
+          {showTxData ? (
+            <>
+              {data}{' '}
+              <LinkWithRef
+                aria-label="Hide details of the transaction"
+                className={classes.linkTxData}
+                onClick={() => setShowTxData(false)}
+                rel="noopener noreferrer"
+                target="_blank"
+              >
+                Show Less
+              </LinkWithRef>
+            </>
+          ) : (
+            <>
+              {shortVersionOf(data, 20)}{' '}
+              <LinkWithRef
+                aria-label="Show details of the transaction"
+                className={classes.linkTxData}
+                onClick={() => setShowTxData(true)}
+                rel="noopener noreferrer"
+                target="_blank"
+              >
+                Show More
+              </LinkWithRef>
+            </>
+          )}
+        </>
+      ) : (
+        data
+      )}
+    </Paragraph>
+  )
+}
+
+interface GenericCustomDataProps {
+  amount?: string
+  data: string
+  recipient: string
+}
+
+const GenericCustomData = ({ amount = '0', data, recipient }: GenericCustomDataProps): React.ReactElement => {
+  const classes = useStyles()
+  const recipientName = useSelector((state) => getNameFromAddressBook(state, recipient))
+
+  return (
+    <Block>
+      <Block data-testid={TRANSACTIONS_DESC_CUSTOM_VALUE_TEST_ID}>
+        <Bold>Send {amount} to:</Bold>
+        {recipientName ? (
+          <OwnerAddressTableCell address={recipient} knownAddress showLinks userName={recipientName} />
+        ) : (
+          <EtherscanLink knownAddress={false} type="address" value={recipient} />
+        )}
+      </Block>
+      <Block className={classes.txData} data-testid={TRANSACTIONS_DESC_CUSTOM_DATA_TEST_ID}>
+        <Bold>Data (hex encoded):</Bold>
+        <TxData data={data} />
+      </Block>
+    </Block>
+  )
+}
+
+interface CustomDescriptionProps {
+  amount?: string
+  data: string
+  recipient: string
+  rawTx: Transaction
+}
+
+const CustomDescription = ({ amount, data, recipient, rawTx }: CustomDescriptionProps): React.ReactElement => {
+  const classes = useStyles()
+
+  return rawTx.multiSendTx ? (
+    <Block className={classes.multiSendTxData} data-testid={TRANSACTIONS_DESC_CUSTOM_DATA_TEST_ID}>
+      {extractMultiSendDecodedData(rawTx).txDetails?.map((tx, index) => {
```

Comment:
could we extract this outside of the jsx to improve readability?

Reply:
no sure I manage to do that but review it made me clean up a bit more the code around this